### PR TITLE
CLN: Remove redundant tests for .duplicated and .drop_duplicates in tests/base

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
part of #23877
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

duplicated and drop_duplicates are already thoroughly tested in `tests/indexes` and `tests/series`. I added comments to highlight the redundancy and extended the existing test cases where needed.

### CODE REVIEW

**Comment by WillAyd** [comment]
Same comment as #32484 are you sure this doesn't lose any coverage?

File: pandas/tests/base/test_ops.py [hunk 1]
```diff
@@ -594,108 +594,6 @@ def test_factorize_repeated(self):
...
-                tm.assert_numpy_array_equal(duplicated, expected)
-                assert duplicated.dtype == bool
-                result = idx.drop_duplicates(keep=False)
-                tm.assert_index_equal(result, idx[~expected])
-
-                with pytest.raises(
-                    TypeError,
-                    match=r"drop_duplicates\(\) got an unexpected keyword argument",
-                ):
-                    idx.drop_duplicates(inplace=True)
```
Comment (SaturnFromTitan) [inline pandas/tests/base/test_ops.py]: Now tested in `test_drop_duplicates_inplace` in `tests/indexes/test_common.py`, see below

File: pandas/tests/base/test_ops.py [hunk 2]
```diff
@@ -594,108 +594,6 @@ def test_factorize_repeated(self):
...
-                base = [False] * len(original) + [True, True]
-                base[3] = True
-                base[5] = True
-                expected = np.array(base)
-
-                duplicated = idx.duplicated(keep=False)
-                tm.assert_numpy_array_equal(duplicated, expected)
-                assert duplicated.dtype == bool
-                result = idx.drop_duplicates(keep=False)
-                tm.assert_index_equal(result, idx[~expected])
```
Comment (SaturnFromTitan) [inline pandas/tests/base/test_ops.py]: Tested in `test_drop_duplicates` in `tests/indexes/test_common.py` which was extended and refactored, see below

File: pandas/tests/base/test_ops.py [hunk 3]
```diff
@@ -594,108 +594,6 @@ def test_factorize_repeated(self):
...
-                    continue
-
-                # original doesn't have duplicates
-                expected = np.array([False] * len(original), dtype=bool)
-                duplicated = original.duplicated()
-                tm.assert_numpy_array_equal(duplicated, expected)
-                assert duplicated.dtype == bool
-                result = original.drop_duplicates()
-                tm.assert_index_equal(result, original)
-                assert result is not original
```
Comment (SaturnFromTitan) [inline pandas/tests/base/test_ops.py]: Now tested in `test_drop_duplicates_no_duplicates` in `tests/indexes/test_common.py`, see below

**Comment by SaturnFromTitan** [comment]
@WillAyd I added comments highlighting and linking the redundant tests. Where needed I extended the existing tests in `tests/indexes/test_common.py` or `tests/series/methods/test_drop_duplicates.py`

**Comment by jreback** [comment]
can you rebase, ping on green.

**Comment by SaturnFromTitan** [comment]
@jreback CI is green

**Comment by jreback** [comment]
thanks @SaturnFromTitan 

there might be other opportunities to use the `keep` fixture in the codebase as well.

**Pull request merged**